# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../tse3/gcc43.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur tse3-0.3.1-orig/src/examples/recording/recording.cpp tse3-0.3.1/src/examples/recording/recording.cpp
--- tse3-0.3.1-orig/src/examples/recording/recording.cpp	2009-02-17 21:31:01.000000000 +0000
+++ tse3-0.3.1/src/examples/recording/recording.cpp	2009-02-17 21:38:29.000000000 +0000
@@ -27,6 +27,7 @@
      **************************************************************/
 
 #include <iostream>
+#include <cstdlib>
 
 // Used in step 1
 #include "tse3/Metronome.h"
diff -Nur tse3-0.3.1-orig/src/tse3/MidiFile.cpp tse3-0.3.1/src/tse3/MidiFile.cpp
--- tse3-0.3.1-orig/src/tse3/MidiFile.cpp	2009-02-17 21:31:01.000000000 +0000
+++ tse3-0.3.1/src/tse3/MidiFile.cpp	2009-02-17 21:40:03.000000000 +0000
@@ -30,6 +30,7 @@
 
 #include <fstream>
 #include <string>
+#include <cstring>
 #include <queue>
 #include <math.h>
 
diff -Nur tse3-0.3.1-orig/src/tse3/Serializable.h tse3-0.3.1/src/tse3/Serializable.h
--- tse3-0.3.1-orig/src/tse3/Serializable.h	2009-02-17 21:31:01.000000000 +0000
+++ tse3-0.3.1/src/tse3/Serializable.h	2009-02-17 21:41:26.000000000 +0000
@@ -20,6 +20,7 @@
 #include <iosfwd>
 #include <iomanip>
 #include <cstddef>
+#include <iostream>
 
 namespace TSE3
 {
diff -Nur tse3-0.3.1-orig/src/tse3/TSE2MDL.cpp tse3-0.3.1/src/tse3/TSE2MDL.cpp
--- tse3-0.3.1-orig/src/tse3/TSE2MDL.cpp	2009-02-17 21:31:01.000000000 +0000
+++ tse3-0.3.1/src/tse3/TSE2MDL.cpp	2009-02-17 21:42:09.000000000 +0000
@@ -38,6 +38,7 @@
 #include "tse3/Progress.h"
 
 #include <fstream>
+#include <cstring>
 
 using namespace TSE3;
 
diff -Nur tse3-0.3.1-orig/src/tse3/cmd/Phrase.h tse3-0.3.1/src/tse3/cmd/Phrase.h
--- tse3-0.3.1-orig/src/tse3/cmd/Phrase.h	2009-02-17 21:31:01.000000000 +0000
+++ tse3-0.3.1/src/tse3/cmd/Phrase.h	2009-02-17 21:32:34.000000000 +0000
@@ -29,6 +29,7 @@
     class Phrase;
     class Song;
     class Part;
+    class PhraseEdit;
 
     namespace Cmd
     {
diff -Nur tse3-0.3.1-orig/src/tse3/plt/OSS.cpp tse3-0.3.1/src/tse3/plt/OSS.cpp
--- tse3-0.3.1-orig/src/tse3/plt/OSS.cpp	2009-02-17 21:31:01.000000000 +0000
+++ tse3-0.3.1/src/tse3/plt/OSS.cpp	2009-02-17 21:40:49.000000000 +0000
@@ -21,6 +21,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <cstring>
 #include <unistd.h>
 #include <fcntl.h>
 #include <stdarg.h>
diff -Nur tse3-0.3.1-orig/src/tse3play/tse3play.cpp tse3-0.3.1/src/tse3play/tse3play.cpp
--- tse3-0.3.1-orig/src/tse3play/tse3play.cpp	2009-02-17 21:31:01.000000000 +0000
+++ tse3-0.3.1/src/tse3play/tse3play.cpp	2009-02-17 21:42:48.000000000 +0000
@@ -32,6 +32,7 @@
 #include "tse3/Error.h"
 #include "tse3/Metronome.h"
 #include <fstream>
+#include <cstdlib>
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
